clients/src/main/java/org/apache/kafka/clients/consumer/CloseOptions.java

```diff
@@ -0,0 +1,124 @@
+     *   </li>
+     * </ul>
+     */
+    public enum GroupMembershipOperation {
+        LEAVE_GROUP,
+        REMAIN_IN_GROUP,
+        DEFAULT
+    }
+
+    /**
+     * Specifies the group membership operation upon shutdown.
+     * By default, {@code GroupMembershipOperation.DEFAULT} will be applied, which follows the consumer's default behavior.
+     */
+    protected GroupMembershipOperation operation = GroupMembershipOperation.DEFAULT;
+
+    /**
+     * Specifies the maximum amount of time to wait for the close process to complete.
+     * This allows users to define a custom timeout for gracefully stopping the consumer.
+     * If no value is set, the default timeout {@link ConsumerUtils#DEFAULT_CLOSE_TIMEOUT_MS} will be applied.
+     */
+    protected Optional<Duration> timeout = Optional.empty();
+
+    private CloseOptions() {
+    }
+
+    protected CloseOptions(final CloseOptions option) {
+        this.operation = option.operation;
+        this.timeout = option.timeout;
+    }
+
+    /**
+     * Static method to create a {@code CloseOptions} with a custom timeout.
+     *
+     * @param timeout the maximum time to wait for the consumer to close.
+     * @return a new {@code CloseOptions} instance with the specified timeout.
+     */
+    public static CloseOptions timeout(final Duration timeout) {
+        CloseOptions option = new CloseOptions();
+        option.timeout = Optional.ofNullable(timeout);
+        return option;
+    }
+
+    /**
+     * Static method to create a {@code CloseOptions} with a specified group membership operation.
+     *
+     * @param operation the group membership operation to apply. Must be one of {@code LEAVE_GROUP}, {@code REMAIN_IN_GROUP},
+     *                  or {@code DEFAULT}.
+     * @return a new {@code CloseOptions} instance with the specified group membership operation.
+     */
+    public static CloseOptions groupMembershipOperation(final GroupMembershipOperation operation) {
+        Objects.requireNonNull(operation, "operation should not be null");
+        CloseOptions option = new CloseOptions();
+        option.operation = operation;
+        return option;
+    }
+
+    /**
+     * Fluent method to set the timeout for the close process.
+     *
+     * @param timeout the maximum time to wait for the consumer to close. If {@code null}, the default timeout will be used.
+     * @return this {@code CloseOptions} instance.
+     */
+    public CloseOptions withTimeout(final Duration timeout) {
+        this.timeout = Optional.ofNullable(timeout);
+        return this;
+    }
+
+    /**
+     * Fluent method to set the group membership operation upon shutdown.
+     *
+     * @param operation the group membership operation to apply. Must be one of {@code LEAVE_GROUP}, {@code REMAIN_IN_GROUP}, or {@code DEFAULT}.
+     * @return this {@code CloseOptions} instance.
+     */
+    public CloseOptions withGroupMembershipOperation(final GroupMembershipOperation operation) {
```

Comment:
Could you please add unit test for those helpers?

Reply:
This is tracked by https://issues.apache.org/jira/browse/KAFKA-18267
I will file a patch for it soon.